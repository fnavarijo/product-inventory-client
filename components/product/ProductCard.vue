<template>
  <NuxtLink
    :to="`/catalog/${id}`"
    class="group m-auto cursor-pointer w-full"
  >
    <div class="relative pt-card bg-light-gray-400 rounded-md flex justify-center">
      <img
        class="object-cover absolute w-4/5 mx-auto top-1/2 transform -translate-y-1/2 group-hover:w-4/6 transition-width ease-in duration-100"
        :src="image"
        alt=""
      >
      <div class="absolute bottom-0 bg-primary-600 text-white uppercase w-full text-center p-3 opacity-0 rounded-br-md rounded-bl-md group-hover:opacity-100 transition ease-in duration-100">
        Ver
      </div>
    </div>
    <div class="flex flex-col mt-4">
      <span class="uppercase">{{ category }}</span>
      <span class="text-lg text-dark-gray-600 font-bold hover:text-primary-600 transition ease-in duration-100">
        {{ name }}
      </span>
    </div>
  </NuxtLink>
</template>

<script lang="ts">
import Vue from 'vue';

import { ProductContainerBase } from '@/components/product/ProductContainerBase';

export default Vue.extend({
  mixins: [ProductContainerBase],
});
</script>
