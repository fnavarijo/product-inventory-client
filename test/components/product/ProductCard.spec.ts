import { mount, RouterLinkStub } from '@vue/test-utils';

import ProductCard from '@/components/product/ProductCard.vue';

describe('ProductCard component', () => {
  it('should render card with name, category and price', () => {
    const wrapper = mount(ProductCard, {
      propsData: {
        category: 'Shoes - Nike',
        name: 'Nike shoes',
        image: 'http://domain.com/image.png',
        id: '123456',
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    });

    expect(wrapper.find('#category').text()).toBe('Shoes');
    expect(wrapper.text()).toContain('Nike shoes');
    expect(wrapper.find('img').attributes('src')).toBe('http://domain.com/image.png');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
