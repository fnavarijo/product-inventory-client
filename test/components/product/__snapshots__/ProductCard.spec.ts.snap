// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component should render card with name, category and price 1`] = `
<a class="group m-auto cursor-pointer">
  <div class="relative pt-card"><img src="http://domain.com/image.png" alt="" class="object-cover absolute">
    <div class="absolute bottom-0 bg-primary-600 text-white uppercase w-full text-center p-3 opacity-0 group-hover:opacity-100 transition ease-in duration-100">
      Ver
    </div>
  </div>
  <div class="flex flex-col mt-4"><span class="uppercase font-light">Shoes</span> <span class="text-lg text-dark-gray-600 hover:text-primary-600 transition ease-in duration-100">Nike shoes</span></div>
</a>
`;
